java/gazelle/testdata/BUILD.bazel

```diff
@@ -0,0 +1,29 @@
+load("@bazel_gazelle//:def.bzl", "gazelle_binary", "gazelle_generation_test")
+
+# We don't want gazelle to run on this directory or its children, as having BUILD.bazel files would stop the test_data globs from reaching into the directories.
+# gazelle:exclude
+
+gazelle_binary(
+    name = "gazelle",
+    languages = [
+        # Register a fake language which makes timestamps consistent via static initializers.
+        # Otherwise we end up with stderr diffs every time we run the tests beacuse of timestamps.
+        # (Yes, this is a horrible hack).
+        "//java/gazelle/private/testutil/reset_zerolog_timestamps",
+        "@bazel_gazelle//language/proto:go_default_library",
+        "//java/gazelle",
+    ],
+    visibility = ["//visibility:private"],
+)
+
+[
+    gazelle_generation_test(
+        name = file[0:-len("/WORKSPACE")],
+        gazelle_binary = ":gazelle",
+        gazelle_timeout_seconds = 30,
+        test_data = glob(
+            include = [file[0:-len("/WORKSPACE")] + "/**"],
```

Comment:
If I understand correctly, this will not glob sub packages, which may lead to some tricky-to-debug situations.

If we want to keep the invariant that "No testdata repo should contain BUILD files", we should have a linter or write it down in the readme.